ui/src/components/SampleControls/SnapshotControl.jsx

```diff
@@ -0,0 +1,60 @@
+import React, { useCallback, useEffect } from 'react';
+import { Button } from 'react-bootstrap';
+import { useSelector } from 'react-redux';
+
+import { download } from './utils';
+
+import styles from './SampleControls.module.css';
+
+const { fabric } = window;
+
+function SnapshotControl(props) {
+  const { canvas } = props;
+
+  const proposal = useSelector((state) => state.login.selectedProposal);
+
+  const currentSampleName = useSelector(({ queue, sampleGrid }) => {
+    const { currentSampleID } = queue;
+    return currentSampleID
+      ? sampleGrid.sampleList[currentSampleID]?.sampleName
+      : 'no-sample';
+  });
+
+  const imageRatio = useSelector((state) => {
+    const { sourceScale, imageRatio } = state.sampleview;
+    return sourceScale * imageRatio;
+  });
+
+  const takeSnapShot = useCallback(() => {
+    const img = document.querySelector('#sample-img');
+    const fimg = new fabric.Image(img);
+    fimg.scale(imageRatio);
+
+    canvas.setBackgroundImage(fimg);
+    canvas.renderAll();
+
+    const filename = `${proposal}-${currentSampleName}.jpeg`;
+    download(filename, canvas.toDataURL({ format: 'jpeg' }));
+
+    canvas.setBackgroundImage(0);
+    canvas.renderAll();
+  }, [canvas, currentSampleName, imageRatio, proposal]);
+
+  useEffect(() => {
+    // Allow server to trigger snapshots (cf. `serverIO`)
+    window.takeSnapshot = takeSnapShot;
```

Comment:
In `serverIO.js`:

```js
    this.hwrSocket.on('take_xtal_snapshot', (cb) => {
      cb(window.takeSnapshot());
    });
```

Just to check: is the `take_xtal_snapshot` socket event still used? It feels a bit brittle, since the canvas has to be displayed for the snapshot to work, so if the event is received when the user is on the _Sample_ screen, it won't work.

Reply:
Its not really, used and we are preparing a PR together with @walesch-yan that will remove it :)